pkg/build/dockerfile-generator.go

```diff
@@ -143,11 +151,24 @@ func (g *DockerfileGenerator) getBaseDockerfile() ([]string, error) {
 }
 
 func (g *DockerfileGenerator) buildPorterSection() []string {
-	return []string{
-		// Remove the user-provided Porter manifest as the canonical version
-		// will migrate via its location in .cnab
-		`RUN rm $BUNDLE_DIR/porter.yaml`,
+	// The user-provided manifest may be located separate from the build context directory.
+	// Therefore, only add lines to remove it from BUNDLE_DIR if its filepath
+	// contains the current working directory.
+	if strings.Contains(g.ManifestPath, g.Getwd()) {
+		manifestSubpaths := strings.SplitAfterN(g.ManifestPath, g.Getwd(), 2)
```

Comment:
I'm having trouble following what's going on here inside the if clause. Are you trying to go from

ManifestPath: ~/stuff/porter.yaml
pwd: ~/stuff

to

[pwd, relative path to porter.yaml] ?

Reply:
(Yes; but this bit removed now that filepath.Rel is being used)